src/config/HYPRE_config.h.in

```diff
@@ -1,203 +1,234 @@
+#undef HYPRE_RELEASE_BUGS
 
-/*- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
- * MEMORY
- *- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -*/
+/* Date of release */
+#undef HYPRE_RELEASE_DATE
 
-/* Define to 1 if using host memory only */
-#undef HYPRE_USING_HOST_MEMORY
+/* Release name */
+#undef HYPRE_RELEASE_NAME
 
-/* Define to 1 if using device memory without UM */
-#undef HYPRE_USING_DEVICE_MEMORY
+/* Time of release */
+#undef HYPRE_RELEASE_TIME
 
-/* Define to 1 if using unified memory */
-#undef HYPRE_USING_UNIFIED_MEMORY
+/* Version number */
+#undef HYPRE_RELEASE_VERSION
 
-/*- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
- * EXECUTION
- *- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -*/
-/* Define to 1 if executing on device with CUDA */
-#undef HYPRE_USING_CUDA
+/* RS6000 arch */
+#undef HYPRE_RS6000
 
-/* Define to 1 if executing on device with OpenMP  */
-#undef HYPRE_USING_DEVICE_OPENMP
+/* Disable MPI, enable serial codes. */
+#undef HYPRE_SEQUENTIAL
 
-/* Define to 1 if using OpenMP on device [target alloc version] */
-#undef HYPRE_DEVICE_OPENMP_ALLOC
+/* HYPRE using single precision */
+#undef HYPRE_SINGLE
 
-/* Define to 1 if using OpenMP on device [target mapped version] */
-#undef HYPRE_DEVICE_OPENMP_MAPPED
+/* Solaris arch */
+#undef HYPRE_SOLARIS
 
-/* Define to 1 if strictly checking OpenMP offload directives */
-#undef HYPRE_DEVICE_OPENMP_CHECK
+/* Using HYPRE timing routines */
+#undef HYPRE_TIMING
 
-/* Define to 1 if executing on host/device with RAJA */
-#undef HYPRE_USING_RAJA
+/* Using Caliper instrumentation */
+#undef HYPRE_USING_CALIPER
 
-/* Define to 1 if executing on host/device with KOKKOS */
-#undef HYPRE_USING_KOKKOS
+/* CUBLAS being used */
+#undef HYPRE_USING_CUBLAS
 
-/* Define to 1 if using NVIDIA Tools Extension (NVTX) */
-#undef HYPRE_USING_NVTX
+/* CUB allocator being used */
+#undef HYPRE_USING_CUB_ALLOCATOR
+
+/* CUDA being used */
+#undef HYPRE_USING_CUDA
+
+/* HYPRE WITH CUDA STREAMS */
+#undef HYPRE_USING_CUDA_STREAMS
 
-/* Define to 1 if using cuSPARSE */
+/* CURAND being used */
+#undef HYPRE_USING_CURAND
+
+/* cuSPARSE being used */
 #undef HYPRE_USING_CUSPARSE
 
-/* Define to 1 if using CUB */
-#undef HYPRE_USING_CUB_ALLOCATOR
+/* HYPRE WITH DEVICE MEMORY */
+#undef HYPRE_USING_DEVICE_MEMORY
 
-/* Define to 1 if using cuBLAS */
-#undef HYPRE_USING_CUBLAS
+/* Enable OpenMP (>=4.5) device directives */
+#undef HYPRE_USING_DEVICE_OPENMP
 
-/* Define to 1 if using cuRAND */
-#undef HYPRE_USING_CURAND
+/* Have external DSuperLU library. */
+#undef HYPRE_USING_DSUPERLU
 
-/* Define to 1 if using GPU aware MPI */
-#undef HYPRE_WITH_GPU_AWARE_MPI
+/* Using dxml for Blas */
+#undef HYPRE_USING_DXML
 
-/* Define to 1 if using CUDA streams */
-#undef HYPRE_USING_CUDA_STREAMS
+/* Using ESSL for Lapack */
+#undef HYPRE_USING_ESSL
 
-/* Define as follows to set the Fortran name mangling scheme:
```

Comment:
Hi @pbauman .  How can we get comments like these into the auto-generated `HYPRE_config.h.in` file?  It is sometimes helpful to just point people directly at the generated `.h` file instead of making them struggle with configure.  In general, I have tried to have a system in place that allows us to bypass the "magic" of configure in a pinch.